app/Models/User.php

```diff
@@ -540,7 +540,7 @@ public function rank()
 
     public function rankHistories()
     {
-        return $this->hasMany(RankHistory::class);
+        return $this->hasMany("App\Models\RankHistory", 'user_id', 'user_id');
```

Comment:
Seems like the primary key isn't set in RankHistory - returns 

```
>>> $r->delete()
Illuminate\Database\QueryException with message 'SQLSTATE[42S22]: Column not found: 1054 Unknown column 'id' in 'where clause' (SQL: delete from `osu_user_performance_rank` where `id` is null)'
```

Maybe should just add to RankHistory: `protected $primaryKey = 'user_id';`?